src/NuGet.Clients/PackageManagement.VisualStudio/Utility/VsHierarchyUtility.cs

```diff
@@ -37,7 +37,31 @@ public static IVsHierarchy ToVsHierarchy(Project project)
             return hierarchy;
         }
 
-        public static VsHierarchyItem GetHierarchyItemForProject(Project project)
+        public static string GetMSBuildProperty(IVsHierarchy pHierarchy, string name)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            var buildPropertyStorage = pHierarchy as IVsBuildPropertyStorage;
+            string output = null;
+
+            if (buildPropertyStorage != null)
+            {
+                var result = buildPropertyStorage.GetPropertyValue(
+                    name,
+                    string.Empty,
+                    (uint)_PersistStorageType.PST_PROJECT_FILE,
+                    out output);
+
+                if (result != NuGetVSConstants.S_OK || string.IsNullOrWhiteSpace(output))
+                {
+                    return null;
+                }
+            }
+
+            return output;
```

Comment:
Trim whitespace before returning (unless you know IVsBuildPropertyStorage will always do this